src/ort_genai_c.h

```diff
@@ -47,34 +60,154 @@ OGA_EXPORT size_t OGA_API_CALL OgaBufferGetDimCount(const OgaBuffer*);
 OGA_EXPORT OgaResult* OGA_API_CALL OgaBufferGetDims(const OgaBuffer*, size_t* dims, size_t dim_count);
 OGA_EXPORT const void* OGA_API_CALL OgaBufferGetData(const OgaBuffer*);
 
-OGA_EXPORT void OGA_API_CALL OgaDestroySequences(OgaSequences*);
-OGA_EXPORT size_t OGA_API_CALL OgaSequencesCount(const OgaSequences*);
-OGA_EXPORT size_t OGA_API_CALL OgaSequencesGetSequenceCount(const OgaSequences*, size_t sequence_index);
-OGA_EXPORT const int32_t* OGA_API_CALL OgaSequencesGetSequenceData(const OgaSequences*, size_t sequence_index);
+/*
+ * \param[in] sequences OgaSequences to be destroyed.
+ */
+OGA_EXPORT void OGA_API_CALL OgaDestroySequences(OgaSequences* sequences);
 
+/*
+ * \brief Returns the number of sequences in the OgaSequences
+ * \param[in] sequences
+ * \return The number of sequences in the OgaSequences
+ */
+OGA_EXPORT size_t OGA_API_CALL OgaSequencesCount(const OgaSequences* sequences);
+
+/*
+ * \brief Returns the number of tokens in the sequence at the given index
+ * \param[in] sequences
+ * \return The number of tokens in the sequence at the given index
+ */
+OGA_EXPORT size_t OGA_API_CALL OgaSequencesGetSequenceCount(const OgaSequences* sequences, size_t sequence_index);
+
+/*
+ * \brief Returns a pointer to the sequence data at the given index. The number of tokens in the sequence
+ *        is given by OgaSequencesGetSequenceCount
+ * \param[in] sequences
+ * \return The pointer to the sequence data at the given index. The sequence data is owned by the OgaSequences
+ *         and will be freed when the OgaSequences is destroyed. The caller must copy the data if it needs to
+ *         be used after the OgaSequences is destroyed.
+ */
+OGA_EXPORT const int32_t* OGA_API_CALL OgaSequencesGetSequenceData(const OgaSequences* sequences, size_t sequence_index);
+
+/*
+ * \brief Creates a model from the given configuration directory and device type.
+ * \param[in] config_path The path to the model configuration directory.
+ * \param[in] device_type The device type to use for the model.
+ * \param[out] out The created model.
+ * \return OgaResult containing the error message if the model creation failed.
+ */
 OGA_EXPORT OgaResult* OGA_API_CALL OgaCreateModel(const char* config_path, OgaDeviceType device_type, OgaModel** out);
-OGA_EXPORT void OGA_API_CALL OgaDestroyModel(OgaModel*);
 
-OGA_EXPORT OgaResult* OGA_API_CALL OgaGenerate(const OgaModel*, OgaGeneratorParams* generator_params, OgaSequences** out);
+/*
+ * \brief Destroys the given model.
+ * \param[in] model The model to be destroyed.
+ */
+OGA_EXPORT void OGA_API_CALL OgaDestroyModel(OgaModel* model);
 
+/*
+ * \brief Generates sequences of tokens from the model execution based on the given generator params.
+ *        The sequences can be thought of as a 2D array of tokens where the first dimension is the batch size
```

Comment:
We shouldn't think of it as a 2D array. It's really an 'array of token arrays', as we should keep the ability for each one to have a different length (vs padding them all in every case). Also not sure if that's best put in the explanation for OgaSequences.